src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -376,34 +380,29 @@ class Peer(
         listen() // This suspends until the coroutines is cancelled or the socket is closed
     }
 
-    private suspend fun watchSwapInWallet(channels: List<PersistedChannelState>) {
-        // Wallet utxos that are already used in channel funding attempts should be ignored, otherwise we would double-spend ourselves.
-        // If the electrum server we connect to has our channel funding attempts in their mempool, those utxos wouldn't be added to our wallet anyway.
-        // But we cannot rely only on that, since we may connect to a different electrum server after a restart, or transactions may be evicted from their mempool.
-        // Since we don't have an easy way of asking electrum to check for double-spends, we would end up with channels that are stuck waiting for confirmations.
-        // This generally wouldn't be a security issue (only one of the funding attempts would succeed), unless 0-conf is used and our LSP is malicious.
-        val initiallyReservedUtxos: Set<OutPoint> = Helpers.reservedWalletInputs(channels)
+    private suspend fun watchSwapInWallet() {
         swapInWallet.walletStateFlow
             .filter { it.consistent }
-            .fold(initiallyReservedUtxos) { reservedUtxos, wallet ->
+            .collect {
                 val currentBlockHeight = currentTipFlow.filterNotNull().first().first
-                val availableWallet = wallet.withoutReservedUtxos(reservedUtxos).withConfirmations(currentBlockHeight, walletParams.swapInConfirmations)
-                logger.info { "swap-in wallet balance (migration=$isMigrationFromLegacyApp): deeplyConfirmed=${availableWallet.deeplyConfirmed.balance}, weaklyConfirmed=${availableWallet.weaklyConfirmed.balance}, unconfirmed=${availableWallet.unconfirmed.balance}" }
-                val utxos = when {
-                    // When migrating from the legacy android app, we use all utxos, even unconfirmed ones.
-                    isMigrationFromLegacyApp -> availableWallet.all
-                    else -> availableWallet.deeplyConfirmed
-                }
-                if (utxos.balance > 0.sat) {
-                    logger.info { "swap-in wallet: requesting channel using ${utxos.size} utxos with balance=${utxos.balance}" }
-                    input.send(RequestChannelOpen(Lightning.randomBytes32(), utxos))
-                    reservedUtxos.union(utxos.map { it.outPoint })
-                } else {
-                    reservedUtxos
-                }
+                input.send(WalletCommand.TrySwapIn(currentBlockHeight, it, walletParams.swapInConfirmations, isMigrationFromLegacyApp))
             }
     }
 
+    private suspend fun watchLiquidityPolicyChanges() {
```

Comment:
I'd like to unit test that logic, but couldn't figure out how to properly interact with `StateFlow`s in `PeerTest.kt` without creating deadlocks...